src/daft-connect/src/translation/logical_plan/sort.rs

```diff
@@ -0,0 +1,83 @@
+use eyre::{bail, WrapErr};
+use spark_connect::expression::{
+    sort_order::{NullOrdering, SortDirection},
+    SortOrder,
+};
+use tracing::warn;
+
+use crate::translation::{logical_plan::LogicalPlanBuilder, to_daft_expr, to_logical_plan};
+
+pub fn sort(sort: spark_connect::Sort) -> eyre::Result<LogicalPlanBuilder> {
+    let spark_connect::Sort {
+        input,
+        order,
+        is_global,
+    } = sort;
+
+    if let Some(is_global) = is_global {
+        warn!("Ignoring is_global {is_global}; not yet implemented");
+    }
+
+    let Some(input) = input else {
+        bail!("Input is required");
+    };
+
+    let plan = to_logical_plan(*input)?;
+
+    let mut sort_by = Vec::new();
+    let mut descending = Vec::new();
+    let mut nulls_first = Vec::new();
+
+    for o in &order {
+        let SortOrder {
+            child,
+            direction,
+            null_ordering,
+        } = o;
+
+        let Some(child) = child else {
+            bail!("Child is required");
+        };
+
+        let child = to_daft_expr(child)?;
+
+        let direction = SortDirection::try_from(*direction)
+            .wrap_err_with(|| format!("Invalid sort direction: {direction:?}"))?;
+
+        let null_ordering = NullOrdering::try_from(*null_ordering)
+            .wrap_err_with(|| format!("Invalid null ordering: {null_ordering:?}"))?;
+
+        // todo(correctness): is this correct?
+        let is_descending = match direction {
+            SortDirection::Unspecified => {
+                bail!("Unspecified sort direction is not yet supported")
+            }
+            SortDirection::Ascending => false,
+            SortDirection::Descending => true,
+        };
+
+        // todo(correctness): is this correct?
+        let tentative_sort_nulls_first = match null_ordering {
+            NullOrdering::SortNullsUnspecified => {
+                bail!("Unspecified null ordering is not yet supported")
+            }
+            NullOrdering::SortNullsFirst => true,
+            NullOrdering::SortNullsLast => false,
+        };
+
+        // https://github.com/Eventual-Inc/Daft/blob/7922d2d810ff92b00008d877aa9a6553bc0dedab/src/daft-core/src/utils/mod.rs#L10-L19
+        let sort_nulls_first = is_descending;
+
+        if sort_nulls_first != tentative_sort_nulls_first {
+            warn!("Ignoring nulls_first {sort_nulls_first}; not yet implemented");
+        }
```

Comment:
we can skip this check as it's already checked closer to the kernel.

Reply:
I get

```
E           pyspark.errors.exceptions.connect.SparkConnectGrpcException: <_MultiThreadedRendezvous of RPC that terminated with:
E               status = StatusCode.INTERNAL
E               details = "Error in Daft server: Failed to apply sort to logical plan
E           
E           Caused by:
E              0: Failed to apply project to logical plan
E              1: Failed to handle unresolved function
E              2: Unresolved function % not yet supported
E           
E           Location:
E               src/daft-connect/src/translation/expr/unresolved_function.rs:27:14"
E               debug_error_string = "UNKNOWN:Error received from peer ipv4:127.0.0.1:59630 {created_time:"2024-11-25T13:58:38.993515-08:00", grpc_status:13, grpc_message:"Error in Daft server: Failed to apply sort to logical plan\n\nCaused by:\n   0: Failed to apply project to logical plan\n   1: Failed to handle unresolved function\n   2: Unresolved function % not yet supported\n\nLocation:\n    src/daft-connect/src/translation/expr/unresolved_function.rs:27:14"}"
E           >
```

Reply:
I get

```
E           pyspark.errors.exceptions.connect.SparkConnectGrpcException: <_MultiThreadedRendezvous of RPC that terminated with:
E               status = StatusCode.INTERNAL
E               details = "Error in Daft server: Not Yet Implemented: nulls_first is not implemented
E           
E           Location:
E               src/daft-connect/src/op/execute/root.rs:43:34"
E               debug_error_string = "UNKNOWN:Error received from peer ipv4:127.0.0.1:60223 {grpc_message:"Error in Daft server: Not Yet Implemented: nulls_first is not implemented\n\nLocation:\n    src/daft-connect/src/op/execute/root.rs:43:34", grpc_status:13, created_time:"2024-11-25T14:05:20.398781-08:00"}"
E           >


```

if I make `let sort_nulls_first = tentative_sort_nulls_first;`

Reply:
it only errors during execution, not planning though, which is what we want. 

ex
```py
df = daft.from_pydict({
    "a": [1, None, 2]
})
df = daft.sql("""
    SELECT * from df order by a DESC NULLS last;
""")
# this works
df.explain()

# but this errors
df.collect()
```

If you try to execute a plan that actually has nulls first/last, we want it to error during execution **NOT** planning.

Reply:
> I get
> 
> ```
> E           pyspark.errors.exceptions.connect.SparkConnectGrpcException: <_MultiThreadedRendezvous of RPC that terminated with:
> E               status = StatusCode.INTERNAL
> E               details = "Error in Daft server: Not Yet Implemented: nulls_first is not implemented
> E           
> E           Location:
> E               src/daft-connect/src/op/execute/root.rs:43:34"
> E               debug_error_string = "UNKNOWN:Error received from peer ipv4:127.0.0.1:60223 {grpc_message:"Error in Daft server: Not Yet Implemented: nulls_first is not implemented\n\nLocation:\n    src/daft-connect/src/op/execute/root.rs:43:34", grpc_status:13, created_time:"2024-11-25T14:05:20.398781-08:00"}"
> E           >
> ```
> 
> if I make `let sort_nulls_first = tentative_sort_nulls_first;`

is this crashing? if so, then this suggests to me that we need better error handling for when a plan fails during execution.

Reply:
no it doesn't crash

Reply:
ok yea I agree with u let's go your way 👍